pkg/supply/supply.go

```diff
@@ -280,12 +280,19 @@ func (s *Supplier) upload(creds *services.IASCredentials, tlsCfg tlsConfig, root
 	if err != nil {
 		return fmt.Errorf("unable to create AMS client: %s", err)
 	}
+	vcapApp, err := env.LoadVcapApplication(s.Log)
+	if err != nil {
+		return err
+	}
 	u := uploader.Uploader{
 		Log:           s.Log,
 		Root:          path.Join(s.Stager.BuildDir(), rootDir),
 		Client:        client,
 		AMSInstanceID: creds.AmsInstanceID,
-		UserAgent:     fmt.Sprintf("cloud-authorization-buildpack/%s", s.BuildpackVersion),
+		ExtraHeaders: map[string]string{
+			"User-Agent":   fmt.Sprintf("cloud-authorization-buildpack/%s", s.BuildpackVersion),
+			"X-Cf-Appname": vcapApp.ApplicationName,
```

Comment:
Let's use a more generic name that also fits for other use cases, e.g. if we provide some functionality to upload via CAP or as a k8s job. E.g. just remove the _cf_ part? `X-App-Name` or `X-Appname` ?